### PR TITLE
Default address country 🗺️ & Phone prefix ☎️ 

### LINKED ISSUE #8081: Add ability to select a default country for the Address and Phone fields
## Scope & Context

All our opportunities, employees and other contacts are in the US and I would like to make Address field to use the United States as a default country. It's tiresome to change it manually all the time.

## Current behavior

Currently, the Country in the Address field defaults to `null` (but shows _Ascention Island_ for some reason, which is confusing).

## Expected behavior

Ability to select the default country on the field level for Address field. I think it would be also useful to add an ability to use a default State for companies that operate only in a single State.

### PR DESCRIPTION
**Pull request by guillim**
# Default address 🗺️ country & Phone ☎️ country

We add the ability to add a Default address country and a default Phone country for fields in the Data model.

fix #8081

### CODE REVIEW

File: packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemInput.ts [hunk 1]
```diff
@@ -18,7 +19,7 @@ export const formatFieldMetadataItemInput = (
   const label = input.label?.trim();
 
   return {
-    defaultValue: input.defaultValue,
+    defaultValue: computeMetadataDefaultValue(input.defaultValue),
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemInput.ts:22]: computeMetadataEscapedDefaultValue

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/address/components/SettingsDataModelFieldAddressForm.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      name="defaultValue"
+      defaultValue={{
+        ...defaultValueInstance,
+        ...fieldMetadataItemDefaultValue,
+      }}
+      control={control}
+      render={({ field: { onChange, value } }) => {
+        const defaultCountry = value?.addressCountry || '';
+        return (
+          <>
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/address/components/SettingsDataModelFieldAddressForm.tsx:65]: non used template

File: packages/twenty-front/src/modules/settings/data-model/fields/preview/hooks/useFieldPreviewValue.ts [hunk 3]
```diff
@@ -45,6 +46,8 @@ export const useFieldPreviewValue = ({
       return getSelectFieldPreviewValue({ fieldMetadataItem });
     case FieldMetadataType.MultiSelect:
       return getMultiSelectFieldPreviewValue({ fieldMetadataItem });
+    case FieldMetadataType.Address:
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/fields/preview/hooks/useFieldPreviewValue.ts:49]: do we need to do that for Phones too?

File: packages/twenty-front/src/modules/object-record/record-field/utils/computeDraftValueFromFieldValue.ts [hunk 4]
```diff
@@ -42,6 +43,20 @@ export const computeDraftValueFromFieldValue = <FieldValue>({
     } as unknown as FieldInputDraftValue<FieldValue>;
   }
 
+  if (isFieldAddress(fieldDefinition)) {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/utils/computeDraftValueFromFieldValue.ts:46]: need to do that for phones too?

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/phones/components/SettingsDataModelFieldPhonesForm.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            <SettingsOptionCardContentSelect
+              Icon={IconMap}
+              dropdownId="selectDefaultCountryCode"
+              title="Default Country Code"
+              description="The default country code for new phone numbers."
+              value={defaultCountryCode}
+              onChange={(newPhoneCountryCode) =>
+                onChange({
+                  ...value,
+                  primaryPhoneCountryCode: newPhoneCountryCode,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/phones/components/SettingsDataModelFieldPhonesForm.tsx:71]: applyQuotes here (and same in Address)

File: packages/twenty-front/src/pages/settings/data-model/utils/compute-metadata-defaultValue-utils.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { applySimpleQuotesToStringRecursive } from '~/utils/string/applySimpleQuotesToString';
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/data-model/utils/compute-metadata-defaultValue-utils.ts:1]: remove this guy

File: packages/twenty-front/src/utils/string/stripSimpleQuotesFromString.ts [hunk 7]
```diff
@@ -6,3 +6,20 @@ export const stripSimpleQuotesFromString = <Input extends string>(
   (simpleQuotesStringSchema.safeParse(value).success
     ? value.slice(1, -1)
     : value) as Input extends `'${infer Output}'` ? Output : Input;
+
+export const stripSimpleQuotesFromStringRecursive = (obj: any): any => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/utils/string/stripSimpleQuotesFromString.ts:10]: same

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemInput.ts (hunk 1 above)
Comment (guillim) [reply packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemInput.ts]: deleted fct

**Comment by guillim** [comment]
Left to be done :

- [ ] this guy `SettingsOptionCardContentSelect` has an onChange typing issue I did not manage to solve so far
- [ ] `PhonesFieldInput` does not load in table mode the default phone country. Still WIP

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/PhonesFieldInput.tsx [hunk 8]
```diff
@@ -48,33 +50,44 @@ type PhonesFieldInputProps = {
...
-  const { persistPhonesField, hotkeyScope, fieldValue } = usePhonesField();
+  const { persistPhonesField, hotkeyScope, draftValue, fieldDefinition } =
+    usePhonesField();
 
-  const phones = useMemo<{ number: string; countryCode: string }[]>(
+  const phones = useMemo<{ number: string; callingCode: string }[]>(
     () =>
       [
-        fieldValue.primaryPhoneNumber
+        draftValue?.primaryPhoneNumber
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/PhonesFieldInput.tsx:59]: if (isDefined(draftValue))

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/PhonesFieldInput.tsx [hunk 9]
```diff
@@ -48,33 +50,44 @@ type PhonesFieldInputProps = {
...
 
+  const defaultCallingCode =
+    stripSimpleQuotesFromString(
+      fieldDefinition?.defaultValue?.primaryPhoneCountryCode,
+    ) ?? '+1';
+
+  // TODO : improve once we store the real country code
+  const defaultCountry = useCountries().find(
+    (obj) => obj.callingCode === defaultCallingCode,
+  )?.countryCode as CountryCode;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/PhonesFieldInput.tsx:82]: updage CountryCode typage

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**